### ISSUE TITLE
MPT

### ISSUE THREAD

**Issue by zphang** [huggingface/transformers#23174]
### Model description

New LLM for MosaicML, 7B parameters. See: https://www.mosaicml.com/blog/mpt-7b

### Open source status

- [X] The model implementation is available
- [X] The model weights are available

### Provide useful links for the implementation

https://huggingface.co/mosaicml/mpt-7b/tree/main

The model is already implemented in a HF/T compatible way, but has multiple source files for its model implementation, some components that aren't used in this current model, and most importantly, dependencies that aren't normally included in HF (e.g. einops, flash_attn).

Do the HF folks have a view on whether we would want to add those dependencies, or implement a vanilla version based only on existing requirements (in which case, it would arguably be easier to modify an existing LM implementation instead, rather than use MosaicML's implemenetation)?

**Comment by YellowRoseCx** [comment]
Can this model be implemented without the need for executing custom python files? I believe it's a huge security risk to allow these models to run their own python code. Yes it is up to the users to verify it's safety and take responsibility for their own actions; yet in the real world, most people won't scan through the python files that come with the models and the next majority might not understand code enough to verify it's safety.

**Comment by SinanAkkoyun** [comment]
Disregarding HF, does anyone have inference code to run the MPT with FasterTransformer?

**Comment by ArthurZucker** [comment]
We are currently talking about this integration with mosaic ML, hope to have updates on this soon!

**Comment by samhavens** [comment]
> Disregarding HF, does anyone have inference code to run the MPT with FasterTransformer?

Here is a script for converting a HuggingFace MPT checkpoint to FasterTransformer https://github.com/mosaicml/llm-foundry/blob/main/scripts/inference/convert_hf_mpt_to_ft.py

**Comment by ricklentz** [comment]
> We are currently talking about this integration with mosaic ML, hope to have updates on this soon!

@ArthurZucker Curious to know how the talks with MosaicML are going. :-)

**Comment by debackerl** [comment]
Probably need to update the ticket description citing MPT-30B as well :-)

**Comment by codertimo** [comment]
@ArthurZucker Any updates on this issue?

**Comment by ArthurZucker** [comment]
Yes! We didn't receive a proper answer, I'll be taking this over! Will open a pr by tomorrow! 😉
